### ISSUE TITLE
frequency and voltage change 

### ISSUE THREAD

**Issue by faezahafiz** [sibyjackgrove/SolarPV-DER-simulation-utility#4]
1. I tried to work in the given example with adding grid event. If I put Vgrid>1.02 or Vgrid<0.88, the ODE solver fails.
2. I changed the grid frequency in the grid event (e.g. fgrid=59). There was no change in the model.

**Comment by sibyjackgrove** [comment]
Can you provide the exact grid events that produced the ODE solver failure? 
**Low voltage events:**
We have been able to run the model up to 0.1 p.u. grid voltage without convergence issues. However, if very low voltages (< 0.5 p.u.) persist for an extended period of time (>1 s) then convergence issues can occur (due to the inverter physics). However, in actual grid-connected inverters, the protection systems disconnect the inverter from the grid if extreme low voltages persist.

**High voltage events:**
High voltage events up to 1.1 p.u. are okay. However, above that, we have observed convergence issues. This is because the inverter has a rated DC link voltage and steady-state duty cycle. If grid voltage is too high the rated DC link voltage won't be sufficient to maintain the necessary voltage at inverter terminals.

Here are some simulations I did with voltage events. Please let me know if you are able to run these scenarios as well.

Here is a run when the voltage falls to 0.8 at 6.0 s and returns back to 1.0 at 7 s. 
```
events1.add_grid_event(6.0,Vgrid=0.8,Vgrid_angle =0.0,fgrid=60.0)
events1.add_grid_event(7.0,Vgrid=1.0,Vgrid_angle =0.0,fgrid=60.0)
```

![image](https://user-images.githubusercontent.com/25213730/69564656-06564d00-0f79-11ea-9555-1b61e03e5f59.png)

Here is a run when the voltage falls to 0.1 at 6.0 s and returns back to 1.0 at 6.1 s. 
```
events1.add_grid_event(6.0,Vgrid=0.1,Vgrid_angle =0.0,fgrid=60.0)
events1.add_grid_event(6.1,Vgrid=1.0,Vgrid_angle =0.0,fgrid=60.0)
```

![image](https://user-images.githubusercontent.com/25213730/69564962-98f6ec00-0f79-11ea-9733-9b0c42606903.png)

Here is a run when the voltage increases to 1.1 p.u.u at 6.0 s and returns back to 1.0 at 7 s. 
```
events1.add_grid_event(6.0,Vgrid=1.1,Vgrid_angle =0.0,fgrid=60.0)
events1.add_grid_event(7.0,Vgrid=1.0,Vgrid_angle =0.0,fgrid=60.0)
```

![image](https://user-images.githubusercontent.com/25213730/69566423-acf01d00-0f7c-11ea-8468-5b1cba1390da.png)

**Comment by sibyjackgrove** [comment]
> 2. I changed the grid frequency in the grid event (e.g. fgrid=59). There was no change in the model.

Frequency events do have an impact on the model. However, there was a bug in the frequency plotting which was fixed with cb0dd594f1c8104abc29db38d1763a90eab14042.

Closing this issue. Please feel free to reopen if further issues related to this topic are encountered.
